crates/hyperdrive-math/src/short/max.rs

```diff
@@ -433,13 +425,14 @@ impl State {
     ) -> Result<FixedPoint> {
         let checkpoint_exposure_shares =
             FixedPoint::try_from(checkpoint_exposure.max(I256::zero()))? / self.vault_share_price();
-        // FIXME: Check this against solidity; should we be using solvency here or just share reserves?
-        // solvency = share_reserves - long_exposure / vault_share_price - min_share_reserves
-        let solvency = self.calculate_solvency();
-        let guess = self.vault_share_price() * (solvency + checkpoint_exposure_shares);
+        let guess = self.vault_share_price().mul_down(
+            self.share_reserves() + checkpoint_exposure_shares
+                - self.long_exposure().div_down(self.vault_share_price())
+                - self.minimum_share_reserves(),
+        );
         let curve_fee = self.curve_fee() * (fixed!(1e18) - spot_price);
         let gov_curve_fee = self.governance_lp_fee() * curve_fee;
-        Ok(guess / (spot_price - curve_fee + gov_curve_fee))
+        Ok(guess.div_down(spot_price - curve_fee + gov_curve_fee))
```

Comment:
This is fine (and good in a lot of ways), but just to make sure that I understand, is the idea that you are trying to eliminate "magic" so that it's easier to quickly parse the rounding direction?

Reply:
I have been following a philosophy of functions that are supposed to mirror solidity should use that syntax (`div_down`) to make it easy for pattern matching and otherwise should use the more readable sugar (`/`).

This function happens to be a mix of both :-P I'll fix it.